service/matching/taskWriter.go

```diff
@@ -166,7 +167,9 @@ func (w *taskWriter) appendTask(
 			return nil, errShutdown
 		}
 	default: // channel is full, throttle
-		return nil, serviceerror.NewResourceExhausted("Too many outstanding appends to the TaskQueue")
+		return nil, serviceerror.NewResourceExhausted(
+			enumspb.RESOURCE_EXHAUSTED_CAUSE_SERVICE_RATE_LIMIT,
+			"Too many outstanding appends to the TaskQueue")
```

Comment:
```suggestion
			"Too many outstanding appends to the task queue")
```
as soon as you're changing this line.